### PR TITLE
Adding 0x trading

### PR DESCRIPTION
**Pull request by weitianjie2000**

### CODE REVIEW

File: src/config/mainnet.json [hunk 1]
```diff
@@ -7,5 +7,10 @@
     "networkName": "mainnet",
     "note": "0xCFEAead4947f0705A14ec42aC3D44129E1Ef3eD5",
     "notional": "0x1344A36A1B56144C3Bc62E7757377D288fDE0369",
+    "sNOTE": "0x38DE42F4BA8a35056b33A746A6b45bE9B1c3B9d2",
+    "treasury": "0x53144559c0d4a3304e2dd9dafbd685247429216d",
+    "balancerVault": "0xBA12222222228d8Ba445958a75a0704d566BF2C8",
+    "balancerPool": "0x6AE82385f76e3742f89cB46343B169f6ee49dE1b",
```
Comment (weitianjie2000) [inline src/config/mainnet.json:13]: This is a placeholder address.

**Comment by hboisselle** [comment]
Lfg

File: src/Notional.ts [hunk 2]
```diff
@@ -88,6 +91,7 @@ export default class Notional extends TransactionBuilder {
       treasury: new Contract(addresses.treasury, TreasuryManagerABI, signer) as TreasuryManager,
       balancerVault: new Contract(addresses.balancerVault, BalancerVaultABI, signer) as BalancerVault,
       balancerPool: new Contract(addresses.balancerPool, BalancerPoolABI, signer) as BalancerPool,
+      exchangeV3: new Contract(addresses.exchangeV3, ExchangeV3ABI, signer) as ExchangeV3,
```
Comment (jeffywu) [inline src/Notional.ts:94]: Add a null check on this and returned undefined if this is not defined

Comment (weitianjie2000) [reply src/Notional.ts]: done

**Review by jeffywu — APPROVED** [review]

**Pull request merged**